youtube_dl/extractor/infomaniak.py

```diff
@@ -0,0 +1,88 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    parse_duration,
```

Comment:
Other suggestions require these imports: 
```suggestion
from ..compat import compat_str as str
from ..utils import (
    ExtractorError,
    merge_dicts,
    parse_duration,
    str_or_none,
    T,
    traverse_obj,
    url_or_none,
```

Reply:
Good catch. For the single-entry case, I now default to id=share_id if needed, and for multi-entry playlists I filter out entries missing id before returning.